src/ansible_compat/roles.py

```diff
@@ -0,0 +1,35 @@
+"""Wrapper class for Ansible Roles."""
+import re
+from keyword import iskeyword
+
+
+class Role:
+    """Utility class for validating roles."""
+
+    name_re = re.compile(r"[a-z\d_]+")
+
+    def __init__(self, name: str, check: bool = False):
+        """Instantiate an Ansible role."""
+        self.namespace = ""
+        if name.count(".") == 1:
+            self.namespace, self.name = name.split(".")
+        else:
+            self.name = name
+        if check and (name.count(".") > 1 or not self.validate()):
+            raise RuntimeError(f"Invalid role name: {self}")
```

Comment:
I would just leave this code out and "force" users to call the `validate` method on the constructed instance. Calling validate here just forces me to handle bad roles in two places: once I construct the instance and once more when I call validate.